include/oneapi/mkl/dft/detail/dft_ct.hxx

```diff
@@ -25,96 +25,114 @@ ONEMKL_EXPORT dft::detail::commit_impl<prec, dom> *create_commit(
 
 // BUFFER version
 
+template <typename descriptor_type>
+using scalar = typename detail::descriptor_info<descriptor_type>::scalar_type;
+template <typename descriptor_type>
+using fwd = typename detail::descriptor_info<descriptor_type>::forward_type;
+template <typename descriptor_type>
+using bwd = typename detail::descriptor_info<descriptor_type>::backward_type;
+
 //In-place transform
-template <typename descriptor_type, typename data_type>
-ONEMKL_EXPORT void compute_forward(descriptor_type &desc, sycl::buffer<data_type, 1> &inout);
+template <typename descriptor_type>
```

Comment:
Are all the function declarations in this file still needed - aren't they replaced by the virtual methods in the commit impl?

Reply:
These are the ones in the `oneapi::mkl::dft::BACKEND` namespace and the virtual methods all just call the corresponding method (in `src/dft/backends/backend_compute_signature.cxx`), so yes I could just move the implementation to the `commit_impl` classes.
Maintaining the backend specific free functions does have the benefit of easily adding the compile-time dispatch in the future, but maybe things would be simpler without it. (and YAGNI principle, but the work is already done).
At the very least I can look at removing the `compute_forward/compute_backward` declarations from this file, since I don't think that needs to be user facing any more.